connectors/bigtable/flink-connector-gcp-bigtable/src/main/java/com/google/flink/connector/gcp/bigtable/writer/BigtableFlushableWriter.java

```diff
@@ -33,26 +51,82 @@
  * <p>At closing, all records are flushed and clients are closed.
  */
 public class BigtableFlushableWriter {
+    String table;
     BigtableDataClient client;
     Batcher<RowMutationEntry, Void> batcher;
 
-    Sink.InitContext context;
+    List<ApiFuture<Void>> batchFutures = new ArrayList<>();
+
+    private Long totalRecordsBuffer = 0L;
+    private Long totalBytesBuffer = 0L;
+    private Counter numRecordsOutCounter;
+    private Counter numBytesOutCounter;
+    private Counter numOutEntryFailuresCounter;
+    private Counter numBatchFailuresCounter;
+    private Histogram numEntriesPerFlush;
+
+    private static final Logger logger = LoggerFactory.getLogger(BigtableFlushableWriter.class);
+
+    private static final Integer BATCHER_CLOSE_TIMEOUT_SECONDS = 60;
 
     public BigtableFlushableWriter(
-            BigtableDataClient client, Sink.InitContext sinkInitContext, String table) {
+            BigtableDataClient client, WriterInitContext sinkInitContext, String table) {
         this.client = client;
+        this.table = table;
         this.batcher = client.newBulkMutationBatcher(TableId.of(table));
-        this.context = sinkInitContext;
+
+        // Instantiate Metrics
+        this.numRecordsOutCounter =
+                sinkInitContext.metricGroup().getIOMetricGroup().getNumRecordsOutCounter();
+        this.numBytesOutCounter =
+                sinkInitContext.metricGroup().getIOMetricGroup().getNumBytesOutCounter();
+        this.numOutEntryFailuresCounter =
+                sinkInitContext.metricGroup().counter("numOutEntryFailuresCounter");
+        this.numBatchFailuresCounter =
+                sinkInitContext.metricGroup().counter("numBatchFailuresCounter");
+        this.numEntriesPerFlush =
+                sinkInitContext
+                        .metricGroup()
+                        .histogram("numEntriesPerFlush", new DescriptiveStatisticsHistogram(100));
```

Comment:
let's make this 100 a constant with a descriptive name. also, why 100 for window size?

Reply:
The window is how many samples are used for the histogram. I jsut felt 100 was a good number. Would you prefer higher? 1000?

Reply:
depends on the expected order of magnitude of the flush sizes. we dont want too many buckets in the histogram

Reply:
It just stores the values, I think an array of 100 Longs is definitely doable.